### PR TITLE
zephyr: cavs: use zephyr pm, clk and dma glue

### PR DESCRIPTION
**Pull request by juimonen**
Start using zephyr pm_runtime, clk and dma glue code in cavs25 native drivers build. Move the files from ace/lib into zephyr/lib.

### CODE REVIEW

File: zephyr/lib/pm_runtime.c
Comment (tmleman) [inline zephyr/lib/pm_runtime.c:98]: I will verify it, but I think it's safe to delete this. This was already discussed when we have rename these registers, but the topic died later.

**Comment by lgirdwood** [comment]
@juimonen ping ? any update ?

File: zephyr/lib/pm_runtime.c
Comment (juimonen) [reply zephyr/lib/pm_runtime.c]: removed now

**Comment by juimonen** [comment]
updated. actually going to D3 now in tgl.

**Review by lgirdwood — COMMENTED** [review]
Do we have any Zephyr commits as a dependecy prior to merge ?

**Comment by juimonen** [comment]
> Do we have any Zephyr commits as a dependecy prior to merge ?
yes, https://github.com/juimonen/zephyr/tree/pm_move

**Comment by juimonen** [comment]
SOFCI TEST

**Comment by juimonen** [comment]
Finally should go to D3 even with asserts enabled. Cavs path had some arch_timer_disable to mess things up...
related zephyr change: https://github.com/zephyrproject-rtos/zephyr/pull/55667

**Review by lgirdwood — COMMENTED** [review]
LGTM, pls ping when Zephyr part merged and update west commit.

**Review by kv2019i — APPROVED** [review]
Looking good. Need the Zephyr dependency of course, but otherwise looks good. We can potentially remove the old implementation soon for cavs25.

File: zephyr/lib/cpu.c
Comment (lyakh) [inline zephyr/lib/cpu.c:60]: would the diff be smaller if you changed this to `#if !CONFIG_ZEPHYR_NATIVE_DRIVERS` and moved the added code here? With this kind of a change it's difficult to see whether the code id changed or just moved.

File: west.yml
Comment (lyakh) [inline west.yml:46]: maybe keep this PR as draft until that Zephyr PR is merged? To make sure this isn't merged by accident

**Comment by juimonen** [comment]
zephyr side pr merged, let's see the CI results...

**Review by tmleman — APPROVED** [review]

File: zephyr/lib/cpu.c
Comment (juimonen) [reply zephyr/lib/cpu.c:60]: yeah probably. I personally dont like the !CONFIGs as they are really "easy" to misinterpret for not having the ! when reading the code. So I would prefer to have the configs without ! if not absolutely needed. But yes, you are also right that this PR didnt change the code of the other branch at all, just moved it around.

**Comment by kv2019i** [comment]
One failure with capture tests but this was happening last week already:
https://sof-ci.01.org/sofpr/PR7183/build5003/devicetest/index.html

System-PM failures on one DUT in (also seen before this PR).
https://sof-ci.01.org/sofpr/PR7183/build5004/devicetest/index.html

Proceeding with merge.

**Pull request merged**
